shared/lib_utility_rate_equations.cpp

```diff
@@ -173,65 +173,102 @@ void rate_data::init(int num_rec_yearly) {
     dc_hourly_peak = std::vector<ssc_number_t>(m_num_rec_yearly, 0);
 }
 
+bool rate_data::check_for_kwh_per_kw_rate(int units) {
+    return (units == 1) || (units == 3);
+}
+
 void rate_data::init_energy_rates(bool gen_only) {
 	// calculate the monthly net energy per tier and period based on units
 	for (int m = 0; m < (int)m_month.size(); m++)
 	{
 		// check for kWh/kW
-        size_t start_tier = 0;
-        size_t end_tier = (int)m_month[m].ec_tou_ub.ncols() - 1;
         size_t num_periods = (int)m_month[m].ec_tou_ub.nrows();
-        size_t num_tiers = end_tier - start_tier + 1;
+        size_t num_tiers = (int)m_month[m].ec_tou_ub.ncols(); // Likely to be overwritten later
 
 		if (!gen_only) // added for two meter no load scenarios to use load tier sizing
 		{
-			//start_tier = 0;
-			end_tier = (int)m_month[m].ec_tou_ub_init.ncols() - 1;
-			//int num_periods = (int)m_month[m].ec_tou_ub_init.nrows();
-			num_tiers = end_tier - start_tier + 1;
-
 
 			// kWh/kW (kWh/kW daily handled in Setup)
 			// 1. find kWh/kW tier
-			// 2. set min tier and max tier based on next item in ec_tou matrix
-			// 3. resize use and chart based on number of tiers in kWh/kW section
+			// 2. fill in kWh tiers as needed, up to kWh/kW * flat peak
+            // 3. repeat until maximum tier is reached
 			// 4. assumption is that all periods in same month have same tier breakdown
 			// 5. assumption is that tier numbering is correct for the kWh/kW breakdown
 			// That is, first tier must be kWh/kW
+            // See example at: https://github.com/NREL/SAM-documentation/blob/master/Unit%20Testing/Utility%20Rates/block_step/GPC_PLL_Tiered_Bill_Calc_Example_v3_btm_tests.xlsx
 			if ((m_month[m].ec_tou_units.ncols() > 0 && m_month[m].ec_tou_units.nrows() > 0)
-				&& ((m_month[m].ec_tou_units.at(0, 0) == 1) || (m_month[m].ec_tou_units.at(0, 0) == 3)))
+				&& check_for_kwh_per_kw_rate(m_month[m].ec_tou_units.at(0, 0)))
```

Comment:
Thanks for breaking this out to work with the new URDB structure that can have two rows of kWh/kW blocks adjacent to one another now.